### ISSUE TITLE
terminated_description: "agreement never appeared on the blockchain"

### ISSUE THREAD

**Issue by linggao** [open-horizon/anax#63]
I did a cancellation tests on an x86 machine over the weekend (66 hours) where the script canceled the agreements every 10 minutes.  There were 2 services registered at any time (location and netspeed). However, when I came back and checked the archived agreements, I saw there were 313 agreements got canceled due to "agreement never appeared on the blockchain", which is every 12 minutes.

**Comment by dabooz** [comment]
Investigating 6 instances of this in a span of 156 tries on the new 1.5.7 blockchain. All 6 did actually appear on the blockchain. Client side timeout should be 5 mins.
1) 06ed6a97ee00...did appear on the blockchain in block 0174400 (0x2A940). This block was received by device geth at 04:15:45 with 1 tx in it. Producer cancellation appeared in block 0174426 approx 6-7 mins after the write by tok02 agbot on staging. device created agreement at 1484712896 (01/18/2017 @ 4:14am UTC), terminated at 1484713418 (01/18/2017 @ 4:23am UTC).  The agbot detected the blockchain write at 1484713013 (01/18/2017 @ 4:16am UTC), anax device never detected the write. Also, the data verification whisper messages are not being delivered to anax at this time, first send was at 04:17:23.

eth filters  04:15:52  fromBlock:0x2a93b toBlock:0x2a93d
04:23:49 fromBlock:0x2a93e toBlock:0x2a93f
04:25:05  fromBlock:0x2a940 toBlock:0x2a94c <== Creation event (and detail event) was in the response to this get logs and so was the producer cancel event. How can the cancel be so close?

Governance worker didnt do anything useful for a long time....
 04:16:51 - resource governance
04:26:48 - resource gov (which fails)
04:26:48 - start processing events

Why so late? 8 min gap between event filter retrievals

**Comment by dabooz** [comment]
So what's the outcome of this issue? I scanned our code to ensure that we were proactively allowing the go dispatcher to dispatch other go routines. I did not find anything that needed to be fixed. The system actually behaved correctly given the sequence of events that occurred.

I did discover that the anax devices (x86 VMs in SL) were configured with 1 CPU, and we know that when geth gets into trouble syncing blocks, it can consume an entire CPU while recovering. Despite the CPU constraints, the horizon code held up. While the single CPU was a somewhat artificial constraint, it's still a good test.

We are wondering if the whisper load contributed to the geth problems (we have no evidence of this other than the fact that no whisper msgs were delivered while geth was in trouble), and thus the feature of adding a message broker to the exchange might have a positive effect on this scenario.
